zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -949,6 +919,70 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
     }
   };
 
+  var getEditorSetting = function(interpreterName) {
+    var deferred = $q.defer();
+    websocketMsgSrv.getEditorSetting(interpreterName);
+    $timeout(
+      $scope.$on('editorSetting', function(event, data) {
+        deferred.resolve(data);
+      }
+    ), 1000);
+    return deferred.promise;
+  };
+
+  var setParagraphMode = function(session, paragraphText, pos) {
+    // Evaluate the mode only if the the position is undefined
+    // or the first 30 characters of the paragraph have been modified
+    // or cursor position is at beginning of second line.(in case user hit enter after typing %magic)
+    if ((typeof pos === 'undefined') || (pos.row === 0 && pos.column < 30) || (pos.row === 1 && pos.column === 0)) {
+      // If paragraph loading, use config value if exists
+      if ((typeof pos === 'undefined') && $scope.paragraph.config.editorMode) {
+        session.setMode($scope.paragraph.config.editorMode);
+      } else {
+        var magic;
+        // set editor mode to default interpreter syntax if paragraph text doesn't start with '%'
+        if (!paragraphText.startsWith('%')) {
+          magic = $scope.$parent.interpreterBindings[0].group;
+        } else {
+          var replNameRegexp = /%(.+?)\s/g;
```

Comment:
This seems like capture the whitespace after any character as well, eg.
it matches `%1 ` for capture group 1 - notice the space after `1`.